python_modules/dagster/dagster/_core/definitions/pipeline_base.py

```diff
@@ -89,20 +85,20 @@ def _subset_for_execution(
     ) -> Self:
         if asset_selection:
             return InMemoryPipeline(
-                self._pipeline_def.get_job_def_for_subset_selection(  # type: ignore  # (must be JobDefinition)
+                self._pipeline_def.get_job_def_for_subset_selection(
                     asset_selection=asset_selection
                 ),
                 asset_selection=asset_selection,
             )
-        if self._pipeline_def.is_subset_pipeline:
+        if self._pipeline_def.is_subset_pipeline:  # should never be called
```

Comment:
if it should never be called should we just get rid of it?

Reply:
That was more of a personal note to myself that I forgot to remove when I was trying to understand the subset system. I'll drop the comment and clean up the logic in a follow-up PR that targets the subsetting system.